packages/graphql/src/translate/translate-read.ts

```diff
@@ -124,30 +126,77 @@ function translateRead({ node, context }: { context: Context; node: Node }): [st
+            if (hasLimit) {
+                const sortFields = optionsInput.sort.map((sort) => Object.keys(sort)[0]);
+                const orderByProjection = createProjectionAndParams({
+                    node,
+                    context,
+                    fieldsByTypeName: {
+                        [node.name]: Object.fromEntries(
+                            Object.entries(fieldsByTypeName[node.name]).filter(([key]) => sortFields.includes(key))
+                        ),
+                    },
+                    varName,
+                });
+                [limitOrderProjStr] = orderByProjection;
+                cypherParams = { ...cypherParams, ...orderByProjection[1] };
+            }
+
             const sortArr = optionsInput.sort.reduce((res: string[], sort: GraphQLSortArg) => {
                 return [
                     ...res,
                     ...Object.entries(sort).map(([field, direction]) => {
-                        return `${varName}.${field} ${direction}`;
+                        if (!cypherSort && node.cypherFields.some((f) => f.fieldName === field)) {
+                            cypherSort = true;
+                        }
+                        return `${hasLimit ? limitOrderVarName : varName}.${field} ${direction}`;
                     }),
                 ];
             }, []);
 
-            sortStr = `ORDER BY ${sortArr.join(", ")}`;
+            sortStr = sortArr.length ? `ORDER BY ${sortArr.join(", ")}` : "";
         }
     }
 
-    const cypher = [
-        matchAndWhereStr,
-        authStr,
-        ...(projAuth ? [`WITH ${varName}`, projAuth] : []),
-        ...connectionStrs,
-        ...interfaceStrs,
-        `RETURN ${varName} ${projStr} as ${varName}`,
-        ...(sortStr ? [sortStr] : []),
-        offsetStr,
-        limitStr,
-    ];
+    let cypher: string[] = [];
+
+    if (!hasLimit) {
+        cypher = [
+            matchAndWhereStr,
+            authStr,
+            ...(sortStr && !cypherSort ? [`WITH ${varName}`, sortStr] : []),
+            ...(projAuth ? [`WITH ${varName}`, projAuth] : []),
+            ...connectionStrs,
+            ...interfaceStrs,
+            `RETURN ${varName} ${projStr} as ${varName}`,
+            ...(sortStr && cypherSort ? [sortStr] : []),
+            offsetStr,
+        ];
+    } else {
+        cypher = [
+            "CALL {",
+            matchAndWhereStr,
+            authStr,
+            ...(sortStr && !cypherSort
+                ? [`WITH ${varName}, ${varName} ${limitOrderProjStr} AS ${limitOrderVarName}`, sortStr]
+                : []),
+            ...(projAuth ? [`WITH ${varName}`, projAuth] : []),
+            `RETURN ${varName}`,
+            ...(!cypherSort ? [offsetStr, limitStr] : []),
+            "}",
+            ...connectionStrs,
+            ...interfaceStrs,
+            `RETURN ${varName} ${projStr} as ${varName}`,
+            ...(sortStr && cypherSort
```

Comment:
I don't think this can go *after* the final return?

I was performing a `WITH` sort,

```
WITH this, this { ... } AS this_sort
ORDER BY this_sort.foo
```

Which cause the result set to be sorted in-line.

Reply:
Damn, good spot, I forgot that `sortStr` was prepended with `WITH`.